submodules/privatelink/vpc-endpoint-services.tf

```diff
@@ -0,0 +1,153 @@
+locals {
+  endpoint_services = tomap({ for service in var.vpc_endpoint_services : service.name => service.private_dns })
+
+  listeners = distinct(flatten([
+    for service in var.vpc_endpoint_services : [
+      for port in service.ports : {
+        service  = service.name
+        port     = port
+        cert_arn = service.cert_arn
+      }
+    ]
+  ]))
+}
+
+data "aws_route53_zone" "hosted" {
+  count        = var.route53_hosted_zone_name != null ? 1 : 0
+  name         = var.route53_hosted_zone_name
+  private_zone = false
+}
+
+data "aws_iam_policy_document" "lb_logs" {
+  statement {
+    sid       = "AWSLogDeliveryAclCheck"
+    effect    = "Allow"
+    resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.deploy_id}-lb-logs"]
+    actions   = ["s3:GetBucketAcl"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [local.aws_account_id]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = ["arn:${data.aws_partition.current.partition}:logs:${var.region}:${local.aws_account_id}:*"]
+    }
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid       = "AWSLogDeliveryWrite"
+    effect    = "Allow"
+    resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.deploy_id}-lb-logs/AWSLogs/*"]
+    actions   = ["s3:PutObject"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+      values   = ["bucket-owner-full-control"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [local.aws_account_id]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = ["arn:${data.aws_partition.current.partition}:logs:${var.region}:${local.aws_account_id}:*"]
+    }
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_s3_bucket" "lb_logs" {
```

Comment:
Can we reuse our existing monitoring bucket instead? Just keeps things in one place for now and infosec can just pull from it

Reply:
yes probably I will replace it